src/main/java/seedu/address/model/tag/Subject.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.tag;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.EnumUtil.inEnum;
+
+/**
+ * Represents a Subject in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidSubjectName(String)}
+ */
+public class Subject {
+    enum Subjects {
+        MATH,
+        PHYSIC, CHEM, BIO,
+        HIS, GEO,
+        ENG,
+    }
```

Comment:
Spell out in full:
MATH,
PHYSICS, CHEMISTRY, BIOLOGY,
LITERATURE, HISTORY, GEOGRAPHY,
ENGLISH, MOTHER TONGUE